lib/ExtUtils/MM_Win32.pm

```diff
@@ -395,8 +395,12 @@ sub perl_script {
 }
 
 sub can_dep_space {
-    my $self = shift;
-    1; # with Win32::GetShortPathName
+	require Win32;
+    # can_dep_space via GetShortPathName, if short paths are supported
+    Win32::CreateFile("8dot3test_canary_can_dep_space $$");
```

Comment:
Sorry for not raising this before. The current directory could be read-only at build time. Creating a file in the current directory does not seem to me correct. Please could you replace this by specifically checking the current filesystem for the shortname capability, rather than this approach?

Reply:
AFAIK there is no reliable way to specifically check a file system for the shortname capability without testing the actual behaviour.

Users with elevated privileges can create a handle to (e.g.) \\.\c​: and pass that handle to 
DeviceIoControl() in order to retrieve FILE_FS_PERSISTENT_VOLUME_INFORMATION, from which support or lack thereof can be determined. Users without those privileges - typical non-administrator users - cannot do that.

Would a viable approach be to try to read the shortname of a file that (1) doesn't fit the 8.3 naming convention and (2) comes with EUMM, perhaps _MakeMaker.pm_? If so, happy to amend this PR, but I'd like advice on what relative path construction to pass to Win32::GetShortPathName.

Reply:
Yes, I think we can guarantee `MakeMaker.pm` would be there, and it's a clever approach. Use `File::Spec` to get the relative filename. Please also address the other issues raised.

Reply:
How does this look?

```
sub can_dep_space {
    my ($self) = @_;
    return 0 unless $self->can_load_xs;
    require Win32;
    require File::Spec;
    my ($vol, $dir, $file);
    ($vol, $dir, undef) = File::Spec->splitpath( $INC{'ExtUtils/MakeMaker.pm'} );
    # can_dep_space via GetShortPathName, if short paths are supported
    my $canary = Win32::GetShortPathName( File::Spec->catpath($vol, $dir, 'MakeMaker.pm') );
    (undef,undef,$file) = File::Spec->splitpath(  $canary );
    return ( length $file > 11 ) ? 0 : 1;
}
```

Reply:
This looks fine. I'd nitpick the `my` could be done better (and I've also made the whitespace consistent):

```
sub can_dep_space {
    my ($self) = @_;
    return 0 unless $self->can_load_xs;
    require Win32;
    require File::Spec;
    my ($vol, $dir) = File::Spec->splitpath($INC{'ExtUtils/MakeMaker.pm'});
    # can_dep_space via GetShortPathName, if short paths are supported
    my $canary = Win32::GetShortPathName(File::Spec->catpath($vol, $dir, 'MakeMaker.pm'));
    (undef, undef, my $file) = File::Spec->splitpath($canary);
    return (length $file > 11) ? 0 : 1;
}
```

Reply:
Thanks. I've updated the PR to your version of _can_dep_space_.